src/prog_models/utils/containers.py

```diff
@@ -35,79 +34,147 @@ def __init__(self, keys : list, data : Union[dict, np.array]):
+    def __add__(self, other: "DictLikeMatrixWrapper") -> "DictLikeMatrixWrapper":
+        """
+        add another matrix to the existing matrix
+        """
         return DictLikeMatrixWrapper(self._keys, self.matrix + other.matrix)
 
     def __iter__(self):
+        """
+        creates iterator object for the list of keys
+        """
         return iter(self._keys)
 
     def __len__(self) -> int:
+        """
+        returns the length of key list
+        """
         return len(self._keys)
 
-    def __eq__(self, other : "DictLikeMatrixWrapper") -> bool:
-        if isinstance(other, dict):
-            return list(self.keys()) == list(other.keys()) and (self.matrix == np.array([[other[key]] for key in self._keys])).all()
-        return self.keys() == other.keys() and (self.matrix == other.matrix).all()
+    def __eq__(self, other: "DictLikeMatrixWrapper") -> bool:
+        """
+        Compares two DictLikeMatrixWrappers (i.e. *Containers) or a DictLikeMatrixWrapper and a dictionary
+        """
+        if isinstance(other, dict): #checks that the list of keys for each matrix match
+            list_key_check = (list(self.keys()) == list(other.keys()))  #checks that the list of keys for each matrix are equal
+            matrix_check = (self.matrix == np.array([[other[key]] for key in self._keys])).all()    #checks to see that each row matches
+            return list_key_check and matrix_check
+        list_key_check = self.keys() == other.keys()
+        matrix_check = (self.matrix == other.matrix).all()
+        return list_key_check and matrix_check
 
     def __hash__(self):
+        """
+        returns hash value sum for keys and matrix
+        """
         return hash(self.keys) + hash(self.matrix)
-    
+
     def __str__(self) -> str:
+        """
+        Represents object as string
+        """
         return self.__repr__()
 
     def get(self, key, default=None):
+        """
+        gets the list of values associated with the key given
+        """
         if key in self._keys:
             return self[key]
         return default
 
     def copy(self) -> "DictLikeMatrixWrapper":
+        """
+        creates copy of object
+        """
         return DictLikeMatrixWrapper(self._keys, self.matrix.copy())
 
     def keys(self) -> list:
+        """
+        returns list of keys for container
+        """
         return self._keys
 
     def values(self) -> np.array:
-        if len(self.matrix) > 0 and len(self.matrix[0]) == 1:
-            return np.array([value[0] for value in self.matrix])
-        return self.matrix
+        """
+        returns array of matrix values
+        """
+        if len(self.matrix) > 0 and len(self.matrix[0]) == 1:   #if the first row of the matrix has one value (i.e., non-vectorized)
+            return np.array([value[0] for value in self.matrix])    #the value from the first row
+        return self.matrix_check    #he matrix (vectorized case)
```

Comment:
```suggestion
        return self.matrix    # the matrix (vectorized case)
```

I think this is supposed to be self.matrix. I think this is why the tests are failing